woocommerce/woocommerce
# Add/e2e more variation tests

### All Submissions:

### Changes proposed in this Pull Request:

Closes #30580.

This PR adds more e2e tests around product variations, particularly:

- can bulk-edit variations (I used the `Toggle "Downloadable"` bulk action for this test)
- can delete all variations
- can manually add a variation
- can manage stock at variation level
- can set variation defaults
- can remove a variation

I also renamed the test `can create variable product variations` to `can create variations from all attributes` to distinguish it from `can manually add a variation`.

Note: In the `can bulk-edit variations` test, I initially tried bulk-editing the price of all variations, but for some reason the test couldn't input values on the prompt dialog using the [dialog.accept([promptText])](https://devdocs.io/puppeteer/index#dialogacceptprompttext) command.

### How to test the changes in this Pull Request:

1. Launch the e2e environment.
2. Run `pnpx wc-e2e test:e2e "./woocommerce/plugins/woocommerce/tests/e2e/specs/wp-admin/create-variable-product.test.js"`
3. All these tests should pass:
<img width="734" alt="Screen Shot 2022-01-06 at 6 56 37 PM" src="https://user-images.githubusercontent.com/4509348/148372808-8a68ae98-3d38-4038-8695-2c3a05e137f9.png">
4. In the UI, navigate to the Products page and verify that the test did not leave any test products.



### Other information:


### Changelog entry

> Added: E2E tests on product variations

### FOR PR REVIEWER ONLY:
